src/net/client/xfr.rs

```diff
@@ -0,0 +1,1147 @@
+//! Parsing of AXFR/IXFR response messages for higher level processing.
+//!
+//! This module provides [`XfrResponseProcessor`] which enables you to process
+//! one or more AXFR/IXFR response messages in terms of the high level
+//! [`XfrEvent`]s that they represent without having to deal with the
+//! AXFR/IXFR protocol details.
+use core::iter::Flatten;
+
+use std::fmt::Debug;
+
+use bytes::Bytes;
+use tracing::trace;
+
+use crate::base::iana::Opcode;
+use crate::base::message::AnyRecordIter;
+use crate::base::wire::ParseError;
+use crate::base::{Message, ParsedName, Record, Rtype, Serial};
+use crate::rdata::{AllRecordData, Soa};
+
+//------------ XfrRecord ------------------------------------------------------
+
+/// The type of record processed by [`XfrResponseProcessor`].
+pub type XfrRecord =
+    Record<ParsedName<Bytes>, AllRecordData<Bytes, ParsedName<Bytes>>>;
+
+//------------ XfrResponseProcessor -------------------------------------------
+
+/// An AXFR/IXFR response processor.
```

Comment:
I wonder why this doesn't use a callback. It seems that the caller has to feed the individual response messages into this, then for each message loop over an iterator to do something, moving to the next  response when the iterator is empty.

Reply:
It was callback but was changed to be an iterator. As that change has now been reverted this comment is now outdated. Resolving.